wasm/store.go

```diff
@@ -416,13 +416,21 @@ func (s *Store) buildFunctionInstances(module *Module, target *ModuleInstance) (
 			tableDeclarations = append(tableDeclarations, imp.Desc.TableTypePtr)
 		}
 	}
+	importedFunctionNum := len(functionDeclarations)
 	functionDeclarations = append(functionDeclarations, module.FunctionSection...)
 	for _, g := range module.GlobalSection {
 		globalDecalarations = append(globalDecalarations, g.Type)
 	}
 	memoryDeclarations = append(memoryDeclarations, module.MemorySection...)
 	tableDeclarations = append(tableDeclarations, module.TableSection...)
 
+	functionNames, _ := module.GetFunctionNames()
+	if functionNames == nil {
+		// We cannot guarantiee the existence of "name" custom section
```

Comment:
also add why "name" could be missing? Ex not in standard yet vs not in the standard we support vs will always be custom. If always custom, some mention of de'facto use since we are supporting it here.

```suggestion
		// We cannot guarantee the existence of "name" custom section
```

Reply:
will add the following: 

```
// That is because the custom section is optional
// in the sepc and some compiler optimize it out to reduce
// the binary size.
```